### PR TITLE
zencoder: add hls support (close #111)

### LINKED ISSUE #111: zencoder: support hls outputs
PR #108 doesn't cover this yet. I'll work on a separate pull request for this.

### PR DESCRIPTION
**Pull request by flavioribeiro**

### CODE REVIEW

File: provider/zencoder/zencoder.go [hunk 1]
```diff
@@ -20,13 +20,14 @@ import (
...
 	"strconv"
+	"strings"
 	"time"
 
 	"github.com/NYTimes/video-transcoding-api/config"
 	"github.com/NYTimes/video-transcoding-api/db"
 	"github.com/NYTimes/video-transcoding-api/db/redis"
 	"github.com/NYTimes/video-transcoding-api/provider"
-	"github.com/brandscreen/zencoder"
+	"github.com/flavioribeiro/zencoder"
```
Comment (flavioribeiro) [inline provider/zencoder/zencoder.go]: if @brandscreen members accept https://github.com/brandscreen/zencoder/pull/4 we can move back to `brandscreen/zencoder` lib

File: provider/zencoder/zencoder.go [hunk 2]
```diff
@@ -89,11 +91,41 @@ func (z *zencoderProvider) buildOutputs(job *db.Job, transcodeProfile provider.T
...
+		outputsWithHLS[len(zencoderOutputs)] = &hlsPlaylist
+		zencoderOutputs = outputsWithHLS
+	}
 	return zencoderOutputs, nil
 }
 
+func (z *zencoderProvider) buildHLSPlaylist(outputs []*zencoder.OutputSettings, hlsOutputs int) zencoder.OutputSettings {
+	output := zencoder.OutputSettings{
+		BaseUrl:  outputs[0].BaseUrl,
+		Filename: "master.m3u8",
```
Comment (fsouza) [inline provider/zencoder/zencoder.go]: The master playlist file name is currently defined by ``TranscodeProfile.StreamingParams.PlaylistFileName``. Not sure if this is the best approach, but we should make it the same across all providers :D

**Comment by fsouza** [comment]
@flavioribeiro can you take a look at the build failure? (the supermaster thing is annoying, but you can probably fool gometalinter by using "super-master" instead of supermaster).

**Comment by flavioribeiro** [comment]
yep sure

**Pull request merged**